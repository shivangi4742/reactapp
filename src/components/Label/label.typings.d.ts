
export type PropTypes = {
    status?: string
}