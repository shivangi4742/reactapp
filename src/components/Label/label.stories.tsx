import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { Grid } from '@material-ui/core';
import Label from "./label";
import ComponentAPI from '../component-api';
import ComponentStories from '../component-stories';
const stories = storiesOf('Components', module);

const codeString = `
  <Label status="active">
    </Label>`
    ;
const codeString1 = `
  <Label status="inactive">
   </Label >
`;
const rows = [{
    id: 0,
    property: 'size',
    description: 'can be set to status as active, inactive, or certified',
    type: 'string',
    default: "-"
}]
stories.add(
    'Label',
    (() => <div>
        <h2> Label </h2>
        Label components are used for collecting user provided information from a list of options.
        <Grid container spacing={16}>

            <ComponentStories syntax={codeString} >
                <Label status="active">
                </Label>
            </ComponentStories>

            <ComponentStories syntax={codeString1} >
                <Label status="inactive">
                </Label>
            </ComponentStories>

        </Grid>
        <ComponentAPI rows={rows} />
    </div>
    )
);