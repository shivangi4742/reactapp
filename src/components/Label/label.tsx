
import React from 'react';
import { LabelWrapper, Labelfont } from './label.styles';
import { PropTypes } from './label.typings';

const Label = ({ status }: PropTypes) => {
    return (
        <LabelWrapper status={status}>
            <Labelfont>{status}</Labelfont>
        </LabelWrapper>
    );

}

export default Label;
