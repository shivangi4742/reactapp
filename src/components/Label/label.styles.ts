
import styled from 'styled-components';
import DS from  '../../utils/design-system';
import { PropTypes } from './label.typings';

export const LabelWrapper = styled.span`
    background-color: ${({ status }: PropTypes) => status === "inactive" ? DS.PALETTE.LIGHT_PINK : status === "active" ? DS.PALETTE.GREEN_LIGHT : DS.PALETTE.BG_LIGHT_BLUE}
    border-radius: 12.5px;
    padding: 6px;
    margin-left: 5px;
    float: left;
` ;
export const Labelfont = styled.span`
   color: #212121;
    font-family: 'robotoregular';
    font-size: 12px;
    font-weight: 400;
    line-height: 14px;
    text-align: center;
    text-transform : capitalize;
`;

