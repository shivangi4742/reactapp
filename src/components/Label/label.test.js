import React from 'react';
import { shallow } from 'enzyme';
import Label from './label';
import toJson from 'enzyme-to-json';
 
describe('Labels',  () => {
    it('should render the label', () => {
        const tree = shallow(<Label status='active'/>);
        expect(toJson(tree)).toMatchSnapshot();
    });
       it('should render the label with different label color', () => {
        const tree = shallow(<Label status='inactive'/>);
        expect(toJson(tree)).toMatchSnapshot();
    });
})