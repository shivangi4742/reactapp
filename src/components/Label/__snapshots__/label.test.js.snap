// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labels should render the label 1`] = `
<styled.span
  status="active"
>
  <styled.span>
    active
  </styled.span>
</styled.span>
`;

exports[`Labels should render the label with different label color 1`] = `
<styled.span
  status="inactive"
>
  <styled.span>
    inactive
  </styled.span>
</styled.span>
`;
